python/lsst/jointcal/jointcal.py

```diff
@@ -679,15 +679,44 @@ def _check_star_lists(self, associations, name):
         if associations.refStarListSize() == 0:
             raise RuntimeError('No stars in the {} reference star list!'.format(name))
 
-    def _logChi2AndValidate(self, associations, fit, model, chi2Label="Model"):
-        """Compute chi2, log it, validate the model, and return chi2."""
+    def _logChi2AndValidate(self, associations, fit, model, chi2Label="Model",
+                            writeChi2Name=None):
+        """Compute chi2, log it, validate the model, and return chi2.
+
+        Parameters
+        ----------
+        associations : `lsst.jointcal.Associations`
+            The star/reference star associations to fit.
+        fit : `lsst.jointcal.FitterBase`
+            The fitter to use for minimization.
+        model : `lsst.jointcal.Model`
+            The model being fit.
+        chi2Label : str, optional
+            Label to describe the chi2 (e.g. "Initialized", "Final").
+        writeChi2Name : `str`, optional
+            Filename to write the chi2 contributions to.
+
+        Returns
+        -------
+        chi2: `lsst.jointcal.Chi2Accumulator`
+            The chi2 object for the current fitter and model.
+
+        Raises
+        ------
+        FloatingPointError
+            Raised if chi2 is infinite or NaN.
+        ValueError
+            Raised if the model is not valid.
```

Comment:
Seems like it'd be desirable to write the diagnostics even (especially) when one of these errors occurs.